File: rust/src/codex/ghost_text.rs
pub struct GhostTextManager;

impl GhostTextManager {
    pub fn new() -> Self {
        GhostTextManager
    }

    pub fn suggest(&self, input: &str, _cursor_position: usize) -> String {
        if input.starts_with("git c") {
            return "ommit -m "update"".to_string();
        }
        if input.starts_with("cargo r") {
            return "un".to_string();
        }
        String::new()
    }

    pub fn render(&self, input: &str, suggestion: &str) -> String {
        format!("{}[90m{}[0m", input, suggestion)
    }
}

File: typescript/src/features/codex/ghost_text.ts
export class GhostTextManager {
  public suggest(input: string, cursorPosition: number): string {
    // Simulating an LLM inline autocomplete
    if (input.startsWith('git c')) {
      return 'ommit -m "update"';
    }
    if (input.startsWith('npm i')) {
      return 'nstall';
    }
    return '';
  }

  public render(input: string, suggestion: string): string {
    // ANSI escape codes: gray color for ghost text
    return `${input}\x1b[90m${suggestion}\x1b[0m`;
  }
}
